File: trunk/src/Client/domain/UserController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Client.communications;
using IServer;
using Client.presentation;
using System.Windows.Forms;

namespace Client.domain{

    public class UserController{

        private static UserController controller;
        private static Ice.Identity callback;
        private String username;

        private UserController(){

        }

        public static UserController getInstance() {
            if (controller == null) {
                controller = new UserController();
            }

            return controller;
        
        }
        
        public void registerUser(String username, String password, String email) {
            try
            {
                ServerProxy.getInstance().registerUser(username, password, email);
            }
            catch (Exception e) 
            {
                MessageBox.Show(e.Message);
            }
        }

        public void loginUser(String username, String password) {
            try
            {
                ServerProxy.getInstance().loginUser(username, password, callback);
                this.Username = username;
                LoginController.getInstance().close();
                WaitingRoomController.getInstance().open();
                
            }
            catch (InvalidLoggingException e)
            {
                switch (e.reason)
                {
                    case "Blocked account":
                        MessageBox.Show("Su cuenta está bloqueada", "Cuenta bloqueada", MessageBoxButtons.OK, MessageBoxIcon.Error);
                        break;
                    case "Incorrect password":
                        MessageBox.Show("La contraseña introducida es incorrecta", "Contraseña incorrecta", MessageBoxButtons.OK, MessageBoxIcon.Error);
                        break;
                    case "Incorrect username":
                        MessageBox.Show("El usuario introducido no existe", "Usuario incorrecto", MessageBoxButtons.OK, MessageBoxIcon.Error);
                        break;
                }
            }
            catch (UserAlreadyLoggedException e)
            {
                MessageBox.Show("Su cuenta está en uso", "Cuenta en uso", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
            catch (Exception e) {
                MessageBox.Show(e.Message, "Cuenta en uso", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }            
        }

        public static void setCallback(Ice.Identity ident) {
            callback = ident;
        }


        public void logoutUser()
        {
            try
            {
                ServerProxy.getInstance().logoutUser(Username);
            }
            catch (Exception e) {
                MessageBox.Show(e.Message, "Cuenta en uso", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
        }

        public String Username
        {
            get { return username; }
            set { username = value; }
        }


    }
}

File: trunk/src/Client/src/domain/UserController.java
package domain;

import communications.Client;
import communications.ServerProxy;
import IServer.InvalidLoggingException;
import IServer.UserAlreadyExistsException;
import IServer.UserAlreadyLoggedException;
import Ice.Identity;
import presentation.LoginUIController;
import presentation.RegisterUIController;

public class UserController {
	private static UserController controller;
	private Identity callback;
	private LoginUIController loginUIController;
	private RegisterUIController registerUIController;
	
	private UserController(){
		loginUIController = new LoginUIController();
	}
	
	public static void initialize() {
		if(controller == null)
			controller = new UserController();
		
	}
	
	public static UserController getInstance(){
		return controller;
	}
	
	public void loginUser(String username, String password) throws InvalidLoggingException, UserAlreadyLoggedException{
		Client.initializeCallback();
		ServerProxy.getProxy().loginUser(username, password, callback);
		loginUIController.close();
		GamesController.initialize();
	}

	public void registerUser(String username, String password, String email) throws UserAlreadyExistsException {
			ServerProxy.getProxy().registerUser(username, password, email);
			loginUIController.activate();
			registerUIController.close();
	}

	public void startRegister() {
		registerUIController = new RegisterUIController();
	}

	public void cancelRegister() {
		loginUIController.activate();
		registerUIController.close();
	}

	/**
	 * @return the callback
	 */
	public Identity getCallback() {
		return callback;
	}

	/**
	 * @param callback the callback to set
	 */
	public void setCallback(Identity callback) {
		this.callback = callback;
	}
	
	
}
